framework/src/preference/private_preference.c

```diff
@@ -276,7 +277,7 @@ int preference_remove_all(void)
 /****************************************************************************
  * Callback Functions
  ****************************************************************************/
-int preference_set_changed_cb(char *key, preference_changed_cb callback, void *user_data)
+int preference_set_changed_cb(const char *key, preference_changed_cb callback, void *user_data)
```

Comment:
Why do you change only `key`?
`Callback` and `user_data` do not need to be changed?

Reply:
The function declaration for `preference_set_changed_cb()` and `preference_unset_changed_cb()` states key as a constant character pointer.
But the current function definitions use a non-constant character pointer.
Hence, I suggest this change.

Reply:
If necessary, you should modify the header as well.

Reply:
The header file `framework/include/preference/preference.h` has a function declaration which identifies the key paramter as a constant character pointer.
However, this header file was not included in `private_preference.c` earlier and the build error indicating conflicting definitions (as shown in description) was avoided.
At the same time, similar function definitions (set_changed_cb and unset_changed_cb) in `shared_preference.c` already use constant keyword for character key pointers.

Hence, I feel adding `const` is correct.

Additionally, why the need for adding the header file (preference/preference.h) now?
This is because since https://github.com/Samsung/TizenRT/pull/5875 we have a missing MACRO error during build, which is also described in the description.

Reply:
I think this public API which is a callback should have constant pointer for key anyways,
but we did not see error because the header file for the API was not included. 
Hence the function definition worked fine as stand alone.

Reply:
Yes,
and done